Max/src/main/java/axi/Broker.java

```diff
@@ -0,0 +1,114 @@
+package axi;
+
+import axi.excepciones.InstrumentoDuplicadoException;
+import axi.excepciones.InstrumentoNoEncontradoException;
+import axi.modelos.InstrumentoFinanciero;
+import axi.modelos.InstrumentoFinancieroFactory;
+import axi.modelos.Inversor;
+import axi.modelos.Tipo;
+
+import java.util.ArrayList;
+
+public class Broker {
+    final static String NAME = "AXI";
+    private static Broker broker;
+    private String name;
+    private ArrayList<InstrumentoFinanciero> instrumentos;
+    private ArrayList<Inversor>inversores;
+
+    private Broker() {
+        this.name = NAME;
+        this.instrumentos = new ArrayList<>();
+        this.inversores = new ArrayList<>();
+    }
+
+    public static Broker getBroker() {
+        if (broker == null)
+            broker = new Broker();
+        return broker;
+    }
+    public void registrarInversor(String nombre, String dni){
+        Inversor inversor = this.buscarInversor(dni);
+        if (inversor != null)
+            throw new RuntimeException("Error. Inversor existente");
+        inversor = new Inversor(nombre, dni);
+        this.inversores.add(inversor);
+    }
+
+    private Inversor buscarInversor(String dni) {
+        Inversor inversor = null;
+        for (int j = 0; j < inversores.size(); j++) {
+            if (inversores.get(j).getDni().equals(dni))
+                inversor = inversores.get(j);
+        }
+        return inversor;
```

Comment:
La utulizar un foreach es mas sencillo de leer el procedimiento de este metodo, si utilizamos un for regular tenemos que indicar con mayor fuerza el valor de la variable que itera

```suggestion
        Inversor inversor = null;
        for (int indice = 0; indice < inversores.size(); indice++) {
            if (inversores.get(indice).getDni().equals(dni))
                inversor = inversores.get(indice);
        }
        return inversor;
```